### ISSUE TITLE
Add option to remove contact from profile screen

### ISSUE THREAD

**Issue by lukaszfryc** [status-im/status-react#3585]
### Description
_Type_: Feature

_Summary_: Before #3570, there was an option to remove a contact from profile screen. Now it's gone and there is no way to remove a contact.

#### Expected behavior
![](https://user-images.githubusercontent.com/7532782/37470718-6f66177a-2868-11e8-950a-432fe8690b86.png)

#### Actual behavior
![](https://user-images.githubusercontent.com/7532782/37470522-feb9391c-2867-11e8-85bf-9fedd70e569a.png)

### Reproduction

* Open Status
* Start new chat
* Open profile of the person
* There should be an menu with option to remove the contact 

### Additional Information

* Status version: latest nightly
* Operating System: Android, iOS
* Issue was introduced in #3570

**Comment by lukaszfryc** [comment]
First we need to specify how "Remove contact" would work. Should a user still se my real name once I removed him/her from my contact list?

**Comment by mandrigin** [comment]
@StatusSceptre can we make this one a bounty?

**Comment by StatusSceptre** [comment]
Yes please add the appropriate labels “bounty-awaiting-approval” and an estimated size for fix. @mandrigin @yenda

**Comment by mandrigin** [comment]
@StatusSceptre ~~added XL label, because it requires changes in both status-go and status-react.~~

UPD: Mixed it up with removing account, changed to `m`.

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 240.0 DAI (240.0 USD @ $1.0/DAI)  attached to it as part of the Status-im fund.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/status-im/status-react/3585/2490).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/status-im/status-react/3585/2490).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $59,437.96 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 12 months from now.
Please review their action plans below:


**1) [eswarasai](https://gitcoin.co/profile/eswarasai) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/status-im/status-react/3585/2490?mutate_worker_action=approve&worker=eswarasai) | [reject worker](https://gitcoin.co/issue/status-im/status-react/3585/2490?mutate_worker_action=reject&worker=eswarasai))_.**

I'll debug the root cause on why the contact menu isn't visible anymore and put a fix for the same. Thanks!

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/status-im/status-react/3585/2490).

**Comment by eswarasai** [comment]
@StatusSceptre @mandrigin -- I'll get started on this one and see if I can quickly resolve this issue. Let me know if there's anything I need to know in advance before jumping into this. Thanks!

**Comment by mandrigin** [comment]
@eswarasai we have a bottom sheet view, so it would be nice if you use it for this task. See `src/status_im/ui/screens/mobile_network_settings/events.cljs:L30`

**Comment by eswarasai** [comment]
@mandrigin -- At the moment, I just tried to get back the old functionality working which utilised `toolbar/actions`. Although the earlier implementation of `Remove from contacts` is basically hiding the user from the `Contacts` sections under `Profile` but if I go to the User Profile via Chat, I can still see that the user is added to my contacts with the `Remove from contacts` option.

I can go ahead and handle to hide the `Remove from contacts` option menu based on the hidden status of the contact. Just wanted to confirm if the above behaviour is the desired one. Thanks :)

**Comment by gitcoinbot** [comment]
@eswarasai Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=1) | [3 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=3) | [5 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=5) | [10 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=10) | [100 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=100)

**Comment by eswarasai** [comment]
@mandrigin -- Any update here? Also, is there any active dev channel that I can join to get instant response for further queries? Thanks :)

**Comment by hesterbruikman** [comment]
@eswarasai Thanks for your patience! Sorry to keep you waiting here. Best channel for more immediate response is:
#status-core http://get.status.im/chat/public/status-core
@StatusSceptre can you share the url to the 'Help' channel for otside of CET hours?

> @mandrigin -- At the moment, I just tried to get back the old functionality working which utilised `toolbar/actions`. Although the earlier implementation of `Remove from contacts` is basically hiding the user from the `Contacts` sections under `Profile` but if I go to the User Profile via Chat, I can still see that the user is added to my contacts with the `Remove from contacts` option.
> 
> I can go ahead and handle to hide the `Remove from contacts` option menu based on the hidden status of the contact. Just wanted to confirm if the above behaviour is the desired one. Thanks :)

Correct, hiding the Remove from contacts option menu is the desired behavior. AFAIK 'actual' contact removal can cause issues that are handled more elegantly by the 'block' user feature. Remove in a way is similar to removing a contact from your phone, it doesn't mean that they can't call you anymore. They are just no longer in your phonebook and you enable 'Caler ID' sort of speak.

Would love a second opinion though, @cammellos is it ok to simply hide 'Remove contact option' on Profile view after it's been used?

**Comment by eswarasai** [comment]
@hesterbruikman -- Thanks for the response. Whatever you've said makes sense. The only confusion is that, even though I've removed the user from the contacts, when I open the user profile, it still shows `In contacts` which can be confusing. I've already hidden the `Remove Contact` menu option. Just wanted to clear out things on the `In contacts` behaviour.

Alternative is to add a check if the user is hidden and display `Add to Contacts` button which will just toggle the value of hide user. Does this make sense to you?

**Comment by hesterbruikman** [comment]
> The only confusion is that, even though I've removed the user from the contacts, when I open the user profile, it still shows `In contacts` which can be confusing.

To see if I understand what's going on. Is this what happens?
- You tap on the [Remove from contacts] when viewing my [Hester] profile
- You go to your [Profile] tab in the bottom menu
- You go to [Contacts]
- You see my name [Hester] in the contact list

**Comment by eswarasai** [comment]
@hesterbruikman -- Sorry for the delayed response. But here's the flow with the screenshots:

1) Tap on the `Remove from contacts` when viewing user profile

<img width="377" alt="screenshot 2019-03-06 at 9 47 57 am" src="https://user-images.githubusercontent.com/5172086/53856414-e3a19400-3ff7-11e9-893d-8f37d92347c0.png">

2) In the `Contacts List` from `Profile` tab in the bottom menu, the user is not visible

<img width="378" alt="screenshot 2019-03-06 at 10 16 39 am" src="https://user-images.githubusercontent.com/5172086/53856675-f49ed500-3ff8-11e9-8f47-43e819132e56.png">

3) But, when we navigate to the user profile from the chat by clicking on the user avatar, below is the screen that appears which can be confusing as it says `In Contacts` but there's no option to `Remove Contact`

<img width="376" alt="screenshot 2019-03-06 at 9 48 11 am" src="https://user-images.githubusercontent.com/5172086/53856418-e8664800-3ff7-11e9-9d70-8916e15593ed.png">


### Note
Another issue I've observed is, if I `Remove Contact`, quit and restart the app, I'm still able to see the `Contact` in the `Contacts List` from `Profile` tab in the bottom menu. I need to dig into this a little more. Just wanted to point it out. Thanks!

**Comment by bitsikka** [comment]
> @StatusSceptre can you share the url to the 'Help' channel for outside of CET hours?

btw there is ***#status-core-help*** for that
https://get.status.im/chat/public/status-core-help

It is where contributors like yourself(@eswarasai) hangout/voluntarily-help-each-other-out/discuss-any-other-concerns. Some with number of bounties and core-dev level experience under their belt.

No guarantees, but, please feel free to ask questions there during off CET hours when core-devs might not be available, or on CET hours and core-devs are way too busy.

Also, the chances of getting answer increases if more and more contributors joins the conversations there. So please join.

**Comment by hesterbruikman** [comment]
@eswarasai thanks for your patience on this! And @bitsikka thanks for jumping in. Indeed, #status-core-help should be a good place to get a faster response to questions.

> Another issue I've observed is, if I `Remove Contact`, quit and restart the app, I'm still able to see the `Contact` in the `Contacts List` from `Profile` tab in the bottom menu. I need to dig into this a little more. Just wanted to point it out. Thanks!

Got some advice from @yenda on what might be going on. There should be a function to delete the contact from the database. If the contact seems to be in contacts after restarting (either in the Contact list or as shown on the user profile image you last shared), it is likely that the contact is not removed from the database.

_Please keep me posted on how this issue is going. We struggle with the semantics of what a contact is and what it means to remove, block, etc. Therefore you might encounter more difficulties or unexpected behavior. If this is a blocker we can freeze this issue for now and invite you back once sorted._

**Comment by eswarasai** [comment]
@hesterbruikman -- No problem. Thanks for getting back to me with the response. I'll spend some more time and try to achieve the above intended behaviour without causing any repercussions, if possible. I'll reach out to you once I have better understanding or hit any blockers :)

**Comment by gitcoinbot** [comment]
@eswarasai Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=1) | [3 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=3) | [5 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=5) | [10 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=10) | [100 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=100)

**Comment by hesterbruikman** [comment]
hi @eswarasai how are you getting on with this issue? if you have any questions, please reach out to https://get.status.im/chat/public/status-core-help

**Comment by yenda** [comment]
@hesterbruikman should we talk about this issue in the profile design meeting tomorrow?

**Comment by gitcoinbot** [comment]
@eswarasai Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=1) | [3 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=3) | [5 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=5) | [10 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=10) | [100 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@eswarasai due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/status-im/status-react/3585/2490) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] reminder (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=1) | [3 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=3) | [5 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=5) | [10 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=10) | [100 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=100)

**Comment by eswarasai** [comment]
@hesterbruikman -- Sorry for the delay in getting back on this issue. 

I've spent some time thinking about the semantics around removing vs blocking contact and also going through the existing implementation/functions. I came across `delete-contact-tx` method which is what you were referring to in the earlier comment, if I'm not wrong but it's not being used anywhere at the moment.

I'm new to Clojurescript and also it's really taking me a while to get my head wrapped around the architecture and the flow. So if this is issue is on priority I can stop work on it and make it available for someone else to pick it up and move it at fast pace. If not, I can try spending more efforts this weekend to get everything in place in order to get the desired behaviour.

Also, I'm curious to know if you guys have discussed about this issue in the Design call and any suggestions/feedback you've got for me in regards with the implementation. Thanks :)

**Comment by gitcoinbot** [comment]
@eswarasai Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=1) | [3 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=3) | [5 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=5) | [10 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=10) | [100 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@eswarasai due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/status-im/status-react/3585/2490) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] reminder (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=1) | [3 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=3) | [5 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=5) | [10 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=10) | [100 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=100)

**Comment by eswarasai** [comment]
@hesterbruikman -- Any update here?

**Comment by yenda** [comment]
@eswarasai as you pointed out the semantics of deleting a contact are complex in this scenario. I think the best solution is to wait for my PR introducing system-tags to be merged https://github.com/status-im/status-react/pull/7805
Deleting a contact would mean removing the  :contact/added tag in that case

**Comment by eswarasai** [comment]
@yenda -- Thanks for the update. That makes sense to me and please do keep me posted once the PR is merged so that I can take this one forward :)

**Comment by gitcoinbot** [comment]
@eswarasai Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=1) | [3 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=3) | [5 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=5) | [10 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=10) | [100 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=100)

**Comment by gitcoinbot** [comment]
@eswarasai Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=1) | [3 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=3) | [5 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=5) | [10 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=10) | [100 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=100)

**Comment by eswarasai** [comment]
@yenda -- Looks like the PR is merged now. Is this issue good to go with? Let me know the exact implementation steps you have in mind so that I can follow them avoiding any further confusion. Thanks :)

**Comment by gitcoinbot** [comment]
@eswarasai Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=1) | [3 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=3) | [5 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=5) | [10 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=10) | [100 days](https://gitcoin.co/issue/status-im/status-react/3585/2490?snooze=100)

**Comment by yenda** [comment]
@hesterbruikman what are your thought on this one based on your work on semantics? The way I see it deleting a contact would be equivalent to removing it from your group of added users, right now we have `:contact/added` system tag that could be removed but `delete` might not be the best term for that

**Comment by hesterbruikman** [comment]
I agree, Delete is not the best term. Deleting a user is simply not possible. Remove from group would be more accurate. 

Following the semantics I'm using now in the [owl ](https://uxenzo.com/~status/userandchatmanagement_1.0.0.owl), there would be an additional tag needed to capture the state of someone who is a Trusted user (a.k.a. added).
:contact/assigned
:group/trusted_users

To capture the state after tapping on Remove in the UI I see 2 options:

- **Option 1**
Assign to a different group that is then specified to not be displayed anywhere. 
:contact/assigned
:group/known (as suggested by @3esmit)
 

- **Option 2**
Expand the language to include an `unassigned` tag. 
:contact/unassigned
_(:group is only relevant when assigned)_

Which do you think is more accurate to what happens? I'm not too sure about option 1 as it feels like keeping a local registry of who you have trusted in the past and I'm not sure if we want a tag to identify these users in case of any leakage.

In case of option 2 I'll add it as Object property in the Turtle file. Will also figure out how to add it to GH one of these days so that changes can start following the same review process as any other updates.

_FYI once we've set on a language, this could be applied to Block user as well, which would change to:
:contact/assigned
:group/blocked_

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 7 months, 1 week from now.
Please review their action plans below:


**1) [ashleyoyt](https://gitcoin.co/profile/ashleyoyt) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/status-im/status-react/3585/2490?mutate_worker_action=approve&worker=ashleyoyt) | [reject worker](https://gitcoin.co/issue/status-im/status-react/3585/2490?mutate_worker_action=reject&worker=ashleyoyt))_.**

Hi, I would like to work on this issue.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/status-im/status-react/3585/2490).

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 7 months, 1 week from now.
Please review their action plans below:


**1) [ashleyoyt](https://gitcoin.co/profile/ashleyoyt) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/status-im/status-react/3585/2490?mutate_worker_action=approve&worker=ashleyoyt) | [reject worker](https://gitcoin.co/issue/status-im/status-react/3585/2490?mutate_worker_action=reject&worker=ashleyoyt))_.**

Hi, I would like to work on this issue.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/status-im/status-react/3585/2490).

**Comment by MarkOSullivan94** [comment]
Can you remove this from Gitcoin? https://gitcoin.co/issue/status-im/status-react/3585/2490

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Cancelled** 

<hr>

__The funding of 240.0 DAI  (240.0 USD @ $1.0/DAI) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://chat.gitcoin.co/'>Gitcoin Chat</a>
* $136,468.60 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
